types.ts

```diff
@@ -79,6 +79,11 @@ export enum AuthenticationType {
    * See https://en.wikipedia.org/wiki/Basic_access_authentication
    */
   WebBasic = 'WebBasic',
+  /**
+   * Authenticate using custom templated values, which you insert into the request. The user provides all the
```

Comment:
Maybe something like:

`Authenticate in a custom way by having one or more arbitrary secret values inserted into the request url or body using template replacement. See {@link CustomAuthentication}.`

Reply:
thoughts on 
```
* Authenticate in a custom way by having one or more arbitrary secret values inserted into the request url, body,
* headers, or the form data using template replacement. See {@link CustomAuthentication}.
```
given im replacing also those pieces in the request?